### ISSUE TITLE
Bug: ref.current always null 

### ISSUE THREAD

**Issue by dev-ahmed** [software-mansion/react-native-reanimated#578]
the issue only happen with flatList

![image](https://user-images.githubusercontent.com/15056155/72907003-d4285800-3d3b-11ea-9d87-5f84ce22cea6.png)

```
import React, { useState, useRef, useEffect } from 'react';
import { View, Text, FlatList, Button } from 'react-native';
import Animated, { Transitioning, Transition } from 'react-native-reanimated';
import PropTypes from 'prop-types';
import styles from './styles';

const AnimatedHorizontalList = ({ data }) => {
  const [list, setList] = useState([]);

  const ref = useRef();

  const transition = (
    <Transition.Sequence>
      <Transition.Out delayMs={200} type="scale" />
      <Transition.Change delayMs={200} interpolation="easeInOut" />
      <Transition.In delayMs={2000} type="fade" />
    </Transition.Sequence>
  );

  useEffect(() => {
      ref.current.animateNextTransition();
      setList(data);
  });

  const renderItem = ({ item }) => {
    return (
      <Transitioning.View ref={ref} transition={transition}>
        <View>
          <Text>{item}</Text>
        </View>
      </Transitioning.View>
    );
  };

  return (
    <View style={styles.container}>
      <FlatList horizontal data={list} renderItem={renderItem} />
    </View>
  );
};

AnimatedHorizontalList.propTypes = {};
export default AnimatedHorizontalList;

```

**Comment by CallMeHeisenberg** [comment]
Okay, so I had similar problem to yours, yet I see to your post is a bit outdated and I hope to still be able to help.

After messing a bit with the debugger I found out to createAnimatedComponent (Probably Transitioning is on the same principle) doesn't directly copy the functions rather under a block called '_component'.

tl;dr your solution **might be**: 
`<Transitioning.View ref={(r) => this.ref = r} transition={transition}>
        <View>
          <Text>{item}</Text>
        </View>
      </Transitioning.View>

useEffect(() => {
      this.ref._component.animateNextTransition();
      setList(data);
  });
`

**Comment by jakub-gonet** [comment]
Hi @dev-ahmed,

1. you don't need `useState` used with `useEffect` here - just use data for rendering data
2. you reuse your `ref` for every element in a list, you need to use an array for that purpose

Try adding a working example on [snack](https://snack.expo.io/).

I'm closing this one, as it doesn't quite report a bug, but feel free to reply.
